app/Controllers/Http/SlackMatchController.js

```diff
@@ -0,0 +1,173 @@
+        return response.status(200).json(message);
+      }
+      if (await Match.taken(time.id)) {      
+        const message = createErrorMessage(`Match at ${time.hour} already taken`);
+        return response.status(200).json(message);
+      }
+
+      await Match.create({
+        user1_id: user1.id,
+        user2_id: user2 ? user2.id : null,
+        time_id: time.id,
+      });
+
+      const message = await matchListMessage();
+      response.status(200).json(message);
+    } catch (error) {
+      const message = createErrorMessage(error.message);
+      response.status(200).json(message);
+    }
+  }
+
+  async matchDelete ({ request, response }) {
+    try {
+      const { text, user_id } = request.body;
+
+      const parameters = text.split(' ');
+      if (parameters.length !== 1) {
+        const message = createErrorMessage('Wrong number of parameters');
+        return response.status(200).json(message);
+      }
+      const [timeHour] = parameters;
+      
+      const time = await Time.findBy('hour', formatHour(timeHour));
+      const user = await User.findBy('slack_id', user_id);
+
+      if (!time) {
+        const message = createErrorMessage(`Time with hour ${timeHour} does not exist`);
+        return response.status(200).json(message);
+      }
+
+      if (!user) {
+        const message = createErrorMessage('Player does not exist \n Try `/playercreate [userName]` before');
+        return response.status(200).json(message);
+      }
+
+      const match = await Match.query().
+        createdToday()
+        .where('time_id', time.id).first();
+
+      if (!match) {
+        const message = createErrorMessage(`Match at ${time.hour} does not exist`);
+        return response.status(200).json(message);
+      }
+
+      if (match.user1_id !== user.id && match.user2_id !== user.id) {
+        const message = createErrorMessage(`You cant delete a match that do not belongs to you :face_with_raised_eyebrow: `);
+        return response.status(200).json(message);
+      }
+
+      await match.delete();
+
+      const message = await matchListMessage();
+      response.status(200).json(message);
+    } catch (error) {
+      response.status(200).json(createErrorMessage(error.message));
+    }
+  }
+
+  async actionsHandler ({ request, response }) {
+    try {
+      const { user, response_url, actions} = JSON.parse(request.body.payload);
+      const matchId = actions[0].value;
+
+      const newPlayer = await User.findBy('slack_id', user.id);
+      if (!newPlayer) {
+        const message = createErrorMessage('Player does not exist \n Try `/playercreate [userName]` before');
+        return response.status(500).json(message);
+      }
+
+      const match = await Match.findOrFail(matchId);
+
+      if (match.user1_id == newPlayer.id) {
+        const message = createErrorMessage(`You cant accept your own challenge`);
+        return response.status(500).json(message);
```

Comment:
never return 500 errors intentionally. The only type of errors you should be showing to the user intentionally is validation (400) errors. 500 codes are all about something you don't anticipate to happen.